src/main/java/frc/robot/subsystems/SwerveModule.java

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.subsystems;
+
+import com.revrobotics.CANSparkMax;
+
+import edu.wpi.first.wpilibj.controller.PIDController;
+import edu.wpi.first.wpilibj.geometry.Rotation2d;
+import edu.wpi.first.wpilibj.geometry.Translation2d;
+import edu.wpi.first.wpilibj.kinematics.SwerveModuleState;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import frc.robot.AbsoluteEncoder;
+import frc.robot.Constants;
+
+public class SwerveModule {
+
+  private CANSparkMax m_driveMotor, m_turningMotor;
+  private AbsoluteEncoder m_turningEncoder;
+  private PIDController m_turningPidController;
+  private Translation2d m_location;
+
+  // Stores current, real state of the wheel, based on information from the
+  // sensors.
+  private SwerveModuleState m_state;
+
+  /**
+   * For the x and y coordinates, forward is along the x-axis. This method also
+   * updates the swerve module state, accessed by getState, for odometry.
+   * 
+   * @param driveMotor     Drive Motor drives the wheel
+   * @param turningMotor   Rotates the wheel
+   * @param X              X Coordinate of the Swerve Wheel
+   * @param Y              Y Coordinate of the Swerve Wheel
+   * @param turningEncoder Measures the angle of the wheel
+   */
+  public SwerveModule(CANSparkMax driveMotor, CANSparkMax turningMotor, double X, double Y,
+      AbsoluteEncoder turningEncoder) {
+    m_driveMotor = driveMotor;
+    m_turningMotor = turningMotor;
+    m_turningEncoder = turningEncoder;
+    m_turningPidController = new PIDController(Constants.SwerveConstants.MODULE_PID_P,
+        Constants.SwerveConstants.MODULE_PID_I, Constants.SwerveConstants.MODULE_PID_D);
+    m_turningPidController.enableContinuousInput(-Math.PI, Math.PI);
+    m_location = new Translation2d(X, Y);
+  }
+
+  public void setDesiredState(SwerveModuleState desiredState) {
+
+    SwerveModuleState state = SwerveModuleState.optimize(desiredState, m_turningEncoder.getRotation2d());
```

Comment:
Not a big issue but imo there should be a clearer difference between state (which is desiredState) and m_state (currentState)

Reply:
I think so too.  I didn't change it because that's how it was before I started working on it.